### PR TITLE
[RemoteMirror] Handle UnsafeContinuation

### PR DESCRIPTION
**Pull request by tbkka**
UnsafeContinuations can be stored in variables or properties, so it's important for RemoteMirror to be able to at least minimally recognize them.

This just treats an UnsafeContinuation as a refcounted pointer. Which might be "good enough" for now.

Resolves rdar://110351406

### CODE REVIEW

File: validation-test/Reflection/reflect_UnsafeContinuation.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// CHECK-NEXT: Instance pointer in child address space: 0x{{[0-9a-fA-F]+}}
+// CHECK-NEXT: Type reference:
+// CHECK-NEXT: (class reflect_UnsafeContinuation.MyClass)
+
+// CHECK-64: Type info:
+// CHECK-64-NEXT: (class_instance size=24 alignment=8 stride=24 num_extra_inhabitants=0 bitwise_takable=1
+// CHECK-64-NEXT:   (field name=cont offset=16
+// CHECK-64-NEXT:     (struct size=8 alignment=8 stride=8 num_extra_inhabitants=2147483647 bitwise_takable=1
+// CHECK-64-NEXT:       (field name=context offset=0
+// CHECK-64-NEXT:         (reference kind=strong refcounting=native)))))
```
Comment (tbkka) [inline validation-test/Reflection/reflect_UnsafeContinuation.swift:49]: I'd love for someone who is familiar with UnsafeContinuation to double-check that this layout looks reasonable.

File: stdlib/public/RemoteInspection/TypeLowering.cpp [hunk 2]
```diff
@@ -1602,6 +1602,12 @@ const TypeInfo *TypeConverter::getDefaultActorStorageTypeInfo() {
   return DefaultActorStorageTI;
 }
 
+const TypeInfo *TypeConverter::getRawUnsafeContinuationTypeInfo() {
+  // FIXME:  This is a bad approximation
+  return getReferenceTypeInfo(ReferenceKind::Strong,
+				 ReferenceCounting::Native);
```
Comment (tbkka) [inline stdlib/public/RemoteInspection/TypeLowering.cpp:1608]: @mikeash Is this even close to right?  Is an UnsafeContinuation just a strong refcounted pointer?

File: validation-test/Reflection/reflect_UnsafeContinuation.swift (hunk 1 above)
Comment (mikeash) [reply validation-test/Reflection/reflect_UnsafeContinuation.swift:49]: I'm not as unfamiliar with the low-level implementation as I probably should be, but from a quick stroll through some relevant compiler sources, it looks like an unsafe continuation is represented as a pointer to the corresponding task. It's not refcounted, exactly, as it borrows a +1 from the code creating it, and transfers that +1 to the task when the continuation is resumed (one reason of many why they must be resumed exactly once!) and isn't retained/released in the meantime, but it is effectively a strong reference during its lifetime.

File: validation-test/Reflection/reflect_UnsafeContinuation.swift [hunk 3]
```diff
@@ -0,0 +1,57 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-build-swift -lswiftSwiftReflectionTest %s -o %t/reflect_UnsafeContinuation
+// RUN: %target-codesign %t/reflect_UnsafeContinuation
+
+// RUN: %target-run %target-swift-reflection-test %t/reflect_UnsafeContinuation | %FileCheck %s --check-prefix=CHECK-%target-ptrsize
+
+// REQUIRES: reflection_test_support
+// REQUIRES: executable_test
+// UNSUPPORTED: use_os_stdlib
```
Comment (mikeash) [inline validation-test/Reflection/reflect_UnsafeContinuation.swift:9]: May also need `UNSUPPORTED: back_deployment_runtime`.

File: stdlib/public/RemoteInspection/TypeLowering.cpp (hunk 2 above)
Comment (mikeash) [reply stdlib/public/RemoteInspection/TypeLowering.cpp:1608]: Answered below, but basically yes. Close enough for our purposes here.

**Review by mikeash — APPROVED** [review]

File: stdlib/public/RemoteInspection/TypeLowering.cpp (hunk 2 above)
Comment (tbkka) [reply stdlib/public/RemoteInspection/TypeLowering.cpp:1608]: Whew!  I'll remove the FIXME then.

File: validation-test/Reflection/reflect_UnsafeContinuation.swift (hunk 3 above)
Comment (tbkka) [reply validation-test/Reflection/reflect_UnsafeContinuation.swift:9]: Good point! Fixed.

File: validation-test/Reflection/reflect_UnsafeContinuation.swift (hunk 1 above)
Comment (tbkka) [reply validation-test/Reflection/reflect_UnsafeContinuation.swift:50]: Thank you for taking a close look.

**Comment by tbkka** [comment]
@swift-ci Please test

**Pull request merged**